amd_openvx_extensions/amd_nn/src/tensor_compare.cpp

```diff
@@ -0,0 +1,287 @@
+    vx_uint32& opencl_local_buffer_size_in_bytes   // [output] reserved: must be ZERO
+)
+{
+    //get tensor dimensions
+    vx_size input_dims[4];
+    vx_size num_of_dims;
+    vx_enum type;
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &num_of_dims, sizeof(num_of_dims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims, sizeof(input_dims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_DATA_TYPE, &type, sizeof(type)));
+
+    strcpy(opencl_kernel_function_name, "tensor_compare");
+
+    opencl_work_dim = 3;
+    opencl_global_work[0] = input_dims[0];
+    opencl_global_work[1] = input_dims[1];
+    opencl_global_work[2] = input_dims[2] * input_dims[3];
+
+    // Setting variables required by the interface
+    opencl_local_buffer_usage_mask = 0;
+    opencl_local_buffer_size_in_bytes = 0;
+
+    if (num_of_dims == 4) {
+        char item[8192];
+        if (type == VX_TYPE_FLOAT32) {
+        sprintf(item,
+                "#pragma OPENCL EXTENSION cl_amd_media_ops : enable\n"
+                "__kernel void %s(__global uchar * in, uint in_offset, uint4 in_stride, __global uchar * in2, uint in2_offset, uint4 in2_stride, __global uchar * out, uint out_offset, uint4 out_stride, uint mode) \n"
+                "{ \n"
+                "     uint x = get_global_id(0);\n"
+                "     uint y = get_global_id(1);\n"
+                "     uint c = get_global_id(2);\n"
+                "     float value = *(__global float *)&in[in_offset + x * in_stride.s0 + y * in_stride.s1 + c * in_stride.s2];\n"
+                "     float value2 = *(__global float *)&in2[in2_offset + x * in2_stride.s0 + y * in2_stride.s1 + c * in2_stride.s2];\n"
+                "     out += out_offset + x  * out_stride.s0 + y * out_stride.s1 + c * out_stride.s2;\n"
+                "     // compare the values and write to the output\n"
+                "     bool result;\n"
+                "     switch (mode) {\n"
+                "          case 0:\n"
+                "            result = (value < value2);\n"
+                "            break;\n"
+                "          case 1:\n"
+                "            result = (value > value2);\n"
+                "            break;\n"
+                "          case 2:\n"
+                "            result = (value <= value2);\n"
+                "            break;\n"
+                "          case 3:\n"
+                "            result = (value >= value2);\n"
+                "            break;\n"
+                "          case 4:\n"
+                "            result = (value == value2);\n"
+                "            break;\n"
+                "          case 5:\n"
+                "            result = (value != value2);\n"
+                "            break;\n"
+                "     }\n"
+                "     *(__global int *)&out[0] = result;\n"
```

Comment:
why are you casting to int here instead of float for the output write?

Reply:
the output value is expecting boolean value (1 & 0)